server/util/networking/networking.go

```diff
@@ -64,8 +66,23 @@ func namespace(netNamespace string, args ...string) []string {
 // executor doesn't exit gracefully. Unfortunately, "ip netns delete" doesn't
 // support patterns, so this runs a short shell for-loop.
 func DeleteNetNamespaces(ctx context.Context) error {
-	return runCommand(ctx, "sh", "-c",
-		"for ns in $(ip netns list | grep ^bb-executor | cut -d \" \" -f 1); do ip netns delete $ns; done")
+	b, err := sudoCommand(ctx, "ip", "netns", "list")
+	if err != nil {
+		return err
+	}
+	output := strings.TrimSpace(string(b))
+	if len(output) == 0 {
+		return nil
+	}
+	for _, ns := range strings.Split(output, "\n") {
+		if !strings.HasPrefix(ns, netNamespacePrefix) {
+			continue
+		}
+		if _, err := sudoCommand(ctx, "ip", "netns", "delete", ns); err != nil {
```

Comment:
Sometimes the output of "ip netns list" can contain spaces, e.g.:

`[07:29:49] iain@lunchbox:~ > ip netns ls
bb-executor-213
bb-executor-21
bb-executor-2
3fe4313e-eb76-4b6d-9d61-53caf12b87e6 (id: 344)
2ab15e85-d1c3-47bc-ad40-74e2941157a4 (id: 332)`

Can you split ns on " " and just delete the first term from the output?

Reply:
Argh, the formatting got all messed up:

[07:29:49] iain@lunchbox:~ > ip netns ls
bb-executor-213
bb-executor-21
bb-executor-2
3fe4313e-eb76-4b6d-9d61-53caf12b87e6 (id: 344)
2ab15e85-d1c3-47bc-ad40-74e2941157a4 (id: 332)